src/libse/Http/HttpClientAdapter.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Nikse.SubtitleEdit.Core.Common;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nikse.SubtitleEdit.Core.Http
+{
+    public class HttpClientAdapter : IHttpClient
+    {
+        private readonly HttpClient _httpClient;
+        private readonly Encoding _encoding;
+
+        public HttpClientAdapter(HttpClient httpClient) : this(httpClient, Encoding.UTF8)
+        {
+        }
+
+        public HttpClientAdapter(HttpClient httpClient, Encoding encoding)
+        {
+            _httpClient = httpClient;
+            // fall back wil always be UTF-8
+            _encoding = encoding ?? Encoding.UTF8;
+        }
+
+        public Uri BaseAddress
+        {
+            get => _httpClient.BaseAddress;
+            set => _httpClient.BaseAddress = value;
+        }
+
+        public HttpRequestHeaders DefaultRequestHeaders => _httpClient.DefaultRequestHeaders;
+
+        public Task<HttpResponseMessage> PostAsync(string uri, StringContent stringContent) => _httpClient.PostAsync(uri, stringContent);
+
+        public async Task<string> GetStringAsync(string url)
+        {
+            var buffer = await _httpClient.GetByteArrayAsync(url);
+            return _encoding.GetString(buffer, 0, buffer.Length);
+            // return Task.FromResult(_encoding.GetString(buffer, 0, buffer.Length - 1));
```

Comment:
` // return Task.FromResult(_encoding.GetString(buffer, 0, buffer.Length - 1));`
I left this line because it was recently introduced and I don't get what is the point of having it?